components/modal/modal.js

```diff
@@ -0,0 +1,238 @@
+import React, { Component, PropTypes } from 'react';
+
+import Global from '../global/global';
+import { getClassNamesWithMods } from '../_helpers';
+
+/**
+ * Modal component
+ */
+class Modal extends Component {
+  state = {
+    isOpen: false,
+  };
+
+  componentDidMount() {
+    if (this.props.closeOnEsc) {
+      global.window.document.addEventListener('keydown', this.handleKeydown);
+    }
+  }
+
+  componentWillReceiveProps(newProps) {
+    if (newProps.active !== this.props.active) {
+      if (newProps.active) {
+        this.open();
+      } else {
+        this.close();
+      }
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.props.closeOnEsc) {
+      global.window.document.removeEventListener('keydown', this.handleKeydown);
+    }
+  }
+
+  open() {
+    global.window.requestAnimationFrame(() => {
+      setTimeout(() => this.setState({ isOpen: true }), 0);
+    });
+  }
+
+  close(e) {
+    if (typeof this.props.onClose === 'function') {
+      this.props.onClose(e);
+    }
+    global.window.requestAnimationFrame(() => {
+      setTimeout(() => this.setState({ isOpen: false }), 300);
+    });
+  }
+
+  handleClose = (e) => {
+    this.close(e);
+  };
+
+  handleKeydown = (e) => {
+    if (e.keyCode === 27) {
+      this.close(e);
+    }
+  };
+
+  handleOverlayClick = (e) => {
+    const { closeOnOverlayClick, onOverlayClick } = this.props;
+    if (typeof onOverlayClick === 'function') {
+      onOverlayClick(e);
+    }
+    if (closeOnOverlayClick) {
+      this.close(e);
+    }
+  };
+
+  renderHeader() {
+    const { title } = this.props;
+    if (!title) {
+      return null;
+    }
```

Comment:
I would separate the `if` conditions with blank lines to have clear indication that is a separate block :)